resilience4j-metrics/src/main/java/io/github/resilience4j/metrics/AsyncRetryMetrics.java

```diff
@@ -36,23 +44,23 @@
 
     private final MetricRegistry metricRegistry = new MetricRegistry();
 
-    private AsyncRetryMetrics(Iterable<AsyncRetry> retries){
+    private AsyncRetryMetrics(Iterable<AsyncRetry> retries) {
         this(DEFAULT_PREFIX, retries);
     }
 
-    private AsyncRetryMetrics(String prefix, Iterable<AsyncRetry> retries){
+    private AsyncRetryMetrics(String prefix, Iterable<AsyncRetry> retries) {
         requireNonNull(prefix);
         requireNonNull(retries);
         retries.forEach(retry -> {
             String name = retry.getName();
 
-            metricRegistry.register(name(prefix, name, SUCCESSFUL_CALLS_WITHOUT_RETRY),
+            metricRegistry.register(name(prefix, name, AsyncRetry.class.getSimpleName(), SUCCESSFUL_CALLS_WITHOUT_RETRY),
```

Comment:
I think we should not automatically append something to the provided name

Reply:
I am afraid we have to to differentiate between sync and async retry metrics otherwise they will override each other if they are used for the back end name together as they will have the same key value at the end

Reply:
Yes I agree, but it should be the prefix. Currently we use DEFAULT_PREFIX = "resilience4j.retry".
We can could add default prefix for AsyncRetry like `resilience4j.asyncRetry.

Reply:
I see your point now :) , i have updated the code accordingly and added extra unit test for micrometer async retry metrics plus resolved the merge conflicts .